File: src/main/java/com/LiveConnect/model/ChatGroup.java
package com.LiveConnect.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@Entity
@NoArgsConstructor
@Table(name = "chat_groups")
public class ChatGroup {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private String name;

    private String description;

    private LocalDateTime creationTime;


    public ChatGroup(String name, String description, LocalDateTime creationTime) {
        this.name = name;
        this.description = description;
        this.creationTime = creationTime;
    }
}

File: src/main/resources/static/javascript/model/chatGroup.js
class ChatGroup {
    constructor({
        id = null,
        name = "",
        description = "",
        creationTime = new Date().toISOString()
    } = {}) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.creationTime = creationTime;
    }

    static fromJSON(json) {
        return new ChatGroup({
            id: json.id,
            name: json.name,
            description: json.description,
            creationTime: json.creationTime,
        });
    }

    toJSON() {
        return {
            id: this.id,
            name: this.name,
            description: this.description,
            creationTime: this.creationTime,
        };
    }
}

export default ChatGroup;
